core/src/banking_stage/unprocessed_transaction_storage.rs

```diff
@@ -149,8 +154,16 @@ fn consume_scan_should_process_packet(
         return ProcessingDecision::Now;
     }
 
-    // Before sanitization, let's quickly check the static keys (performance optimization)
+    // Check if the fee payer is in the invalid fee payer filter
     let message = &packet.transaction().get_message().message;
+    if invalid_fee_payer_filter.should_reject(&message.static_account_keys()[0]) {
+        payload
+            .message_hash_to_transaction
```

Comment:
possible to get this check in front of the map insertion so we don't have to bother with removal here?

Reply:
That's actually already done, but we still need this 😦 

This `message_hash_to_transaction` is a reference to a map that tracks all our pending packets. So any packets that made it into the buffer on receiving will be in the map.

Anytime we make a `Never` decision, i.e. this packet will never be processed, we then remove it from the map.

This PR checks against invalid_fee_payer before we insert into the buffer.
But consider the case where an invalid fee-payer submits a ton of invalid fee-paying txs before our leader slot begins (this was happening recently).
Their packets are already in our buffer, before we can identify their key as unable to pay fees.
The code here is for quickly rejecting packets that are *already* in the buffer.

Kind of a long description, hopefully that makes sense?